test/pylib/scylla_cluster.py

```diff
@@ -560,17 +563,19 @@ def server_log_filename(self) -> Optional[pathlib.Path]:
         """The log file name of the failed server"""
         # FIXME: pick failed server
         if self.running:
-            return next(iter(self.running.values())).log_filename
+            return next(server for server in self.running.values()).log_filename
         else:
             return None
 
     def __str__(self):
-        running = f"{{{', '.join(str(c) for c in self.running)}}}"
-        stopped = f"{{{', '.join(str(c) for c in self.stopped)}}}"
+        running = ", ".join(str(server) for server in self.running.values())
+        stopped = ", ".join(str(server) for server in self.stopped.values())
         return f"ScyllaCluster(name: {self.name}, running: {running}, stopped: {stopped})"
 
-    def running_servers(self) -> List[str]:
-        return list(set(self.running.keys()) - self.removed)
+    def running_servers(self) -> List[Tuple[ServerNum, IPAddress]]:
```

Comment:
Using `ServerInfo` here doesn't work?

Reply:
It's only used to return a JSON via API. The type doesn't help.